nvflare/app_common/executors/learner_executor.py

```diff
@@ -71,7 +76,9 @@ def initialize(self, fl_ctx: FLContext):
 
     def execute(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -> Shareable:
         self.log_info(fl_ctx, f"Client trainer got task: {task_name}")
-
+        if not self.is_initialized:
+            self.is_initialized = True
+            self.initialize(fl_ctx)
```

Comment:
If executor fails to initialize, the task's return code will be EXECUTION_ERROR, which will cause the whole job to be aborted.

Right now, since is_unsafe is only set during task execution (which happens after initialize), there is no need to check this flag in the initialize function.

HOWEVER, we probably should make this a general feature, meaning that this flag could be set by any component at any time. Then you are right - we should check this flag in initialize. Yuhong seems to be suggesting too.

In this case, we should keep is_unsafe flag as a global property, not specific to executors - there could be many other components that could benefit from this flag.

Reply:
Changed to set unsafe_job as a sticky prop in FL Context. When any component (other than filters and executors) detects the job is unsafe, it can call fl_ctx.set_job_is_unsafe() to set the prop. 

The client_runner has been modified to end the current job if the unsafe_job prop is set.